src/main/kotlin/org/radarbase/push/integration/garmin/resource/GarminPushEndpoint.kt

```diff
@@ -0,0 +1,131 @@
+package org.radarbase.push.integration.garmin.resource
+
+import com.fasterxml.jackson.databind.JsonNode
+import org.radarbase.jersey.auth.Authenticated
+import org.radarbase.push.integration.garmin.service.GarminHealthApiService
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import javax.inject.Singleton
+import javax.ws.rs.Consumes
+import javax.ws.rs.POST
+import javax.ws.rs.Path
+import javax.ws.rs.container.ContainerRequestContext
+import javax.ws.rs.core.Context
+import javax.ws.rs.core.MediaType
+import javax.ws.rs.core.Response
+
+@Consumes(MediaType.APPLICATION_JSON)
+@Singleton
+@Path("garmin")
+@Authenticated
+class GarminPushEndpoint(@Context private val healthApiService: GarminHealthApiService) {
+
+    @POST
+    @Path("dailies")
+    fun addDalies(@Context requestContext: ContainerRequestContext): Response {
+        val tree = requestContext.getProperty("tree") as JsonNode
```

Comment:
Since this particular line is repeated quite often. If you intend on avoiding parsing the content twice, you could make a factory `GarminJsonNodeFactory` that lets you inject `@Context @Named("garmin") tree: JsonNode` as a method parameter, and likewise add a GarminUserFactory. The factory just reads it from the request context. You can even add the JsonNode and User as class vals, if you set `.proxy(true)` in the inject binding.

Alternatively, you add `requestContext` as a class val, and create class properties:
```kotlin
val tree: JsonNode
    get() = requestContext.getProperty("tree") as JsonNode
val user: User
    get() = requestContext.getProperty("user") as User
```
In both cases, requestContext can be removed from the method parameters. The converter also probably does not need the request context?

Reply:
Yes, i think we can pass the user instead of requestContext and should be enough for the converters